sentence_transformers/cross_encoder/evaluation/CEF1Evaluator.py

```diff
@@ -0,0 +1,128 @@
+import csv
+import logging
+import os
+from typing import List
+
+import numpy as np
+from ... import InputExample
+from sklearn.metrics import f1_score
+
+logger = logging.getLogger(__name__)
+
+
+class CEF1Evaluator:
+    """
+    CrossEncoder F1 score based evaluator for binary and multiclass tasks.
+
+    The task type (binary or multiclass) is determined from the labels array.
+
+    :param sentence_pairs: A list of sentence pairs, where each pair is a list of two strings.
+    :type sentence_pairs: list[list[str]]
+    :param labels: A list of integer labels corresponding to each sentence pair.
+    :type labels: list[int]
+    :param batch_size: Batch size for prediction. Defaults to 32.
+    :type batch_size: int
+    :param show_progress_bar: Show tqdm progress bar.
+    :type show_progress_bar: bool
+    :param name: An optional name for the CSV file with stored results. Defaults to an empty string.
+    :type name: str, optional
+    :param write_csv: Flag to determine if the data should be saved to a CSV file. Defaults to True.
+    :type write_csv: bool, optional
+    """
+
+    def __init__(
+        self,
+        sentence_pairs: List[List[str]],
+        labels: List[int],
+        *,
+        batch_size: int = 32,
+        show_progress_bar: bool = None,
+        name: str = "",
+        write_csv: bool = True,
+    ):
+        self.sentence_pairs = sentence_pairs
+        self.labels = labels
+        self.batch_size = batch_size
+        self.show_progress_bar = show_progress_bar
+        self.name = name
+        self.write_csv = write_csv
+
+        n_unique = np.unique(labels)
+        if n_unique == 2:
+            problem_type = "binary"
+        elif n_unique > 2:
+            problem_type = "multiclass"
+        else:
+            raise ValueError(
+                "Got only one distinct label. Please make sure there are at least two labels in the `labels` array."
+            )
+
+        if problem_type == "binary":
+            self.f1_callables = [
+                ("Binary F1 score", lambda x, y: f1_score(x, y, average="binary")),
+            ]
+        elif problem_type == "multiclass":
+            self.f1_callables = [
+                ("Macro F1 score", lambda x, y: f1_score(x, y, average="macro")),
+                ("Micro F1 score", lambda x, y: f1_score(x, y, average="micro")),
+                ("Weighted F1 score", lambda x, y: f1_score(x, y, average="weighted")),
+            ]
+
+        self.csv_file = "CEF1Evaluator" + (f"_{name}" if name else "") + "_results.csv"
+        self.csv_headers = ["epoch", "steps"] + [metric_name for metric_name, _ in self.f1_callables]
+
+    @classmethod
+    def from_input_examples(cls, problem_type: str, examples: List[InputExample], **kwargs):
```

Comment:
```suggestion
    def from_input_examples(cls, examples: List[InputExample], **kwargs):
```
I think these are from an earlier version of your code perhaps?

Reply:
Good catch. Thanks.